mcs_benchmark_data/_benchmark_transformer.py

```diff
@@ -62,37 +68,92 @@ def _transform(self, **kwds) -> Generator[_Model, None, None]:
 
             dataset_uri = URIRef(f"{self._uri_base}:dataset:{dataset['@id']}")
 
-            new_dataset = self._benchmark_dataset_classes[dataset_type](
+            new_dataset = self.__benchmark_dataset_classes[dataset_type](
                 uri=dataset_uri, benchmark_uri=benchmark.uri, name=dataset["name"]
             )
 
             yield new_dataset
 
             yield from self._transform_benchmark_sample(
+                extracted_path=extracted_path,
+                file_names=file_names,
                 dataset_type=dataset_type,
                 dataset_uri=new_dataset.uri,
                 **kwds,
             )
 
-    def _prepare_sample(
+    def _yield_qa_models(
         self,
         *,
-        dataset_type: str,
         dataset_uri: URIRef,
-        sample: dict,
-        correct_choice: int,
+        sample_id: str,
+        concepts: Optional[Tuple[str, ...]],
+        context: Optional[str],
+        correct_choice: URIRef,
+        question: str,
+        answers: Tuple[Tuple[str, str], ...],

```

Comment:
Not obvious from looking at answers what its shape really is. Prefer using a NamedTuple or a dataclass for the Tuple[str, str].